File: appcenter-distribute/ios/Plugin/AppCenterDistributePlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(DistributePlugin, "Distribute",
           CAP_PLUGIN_METHOD(checkForUpdate, CAPPluginReturnNone);
           CAP_PLUGIN_METHOD(setEnabled, CAPPluginReturnNone);
           CAP_PLUGIN_METHOD(isEnabled, CAPPluginReturnPromise);
)

File: appcenter-distribute/ios/Plugin/AppCenterDistributePlugin.swift
import Foundation
import Capacitor
import AppCenterCapacitorShared

@objc(DistributePlugin)
public class DistributePlugin: CAPPlugin {
    private let implementation = AppCenterDistributeBase()

    override public func load() {
        print("[DistributePlugin] load")

        AppCenterCapacitorShared.configureWithSettings()

        let config: NSDictionary = AppCenterCapacitorShared.getConfiguration()

        // get Distribute config options
        let disableAutomaticCheckForUpdate = config["DistributeDisableAutomaticCheckForUpdate"] as? Bool
        // DistributeUpdateTrack
        let updateTrack = config["DistributeUpdateTrack"] as? String

        // TODO: use config options
        if AppCenterCapacitorShared.isSdkConfigured() {
            print("[DistributePlugin] starting")
            implementation.start()
        }
    }

    @objc func checkForUpdate(_ call: CAPPluginCall) {
        implementation.checkForUpdate()
        call.resolve()
    }

    @objc func setEnabled(_ call: CAPPluginCall) {
        implementation.enable(call.getBool("enable") ?? false)
        call.resolve()
    }

    @objc func isEnabled(_ call: CAPPluginCall) {
        call.resolve(["value": implementation.isEnabled()])
    }
}
